go/parquet/pqarrow/encode_arrow.go

```diff
@@ -36,19 +36,17 @@ import (
 
 // get the count of the number of leaf arrays for the type
 func calcLeafCount(dt arrow.DataType) int {
+	if dt, ok := dt.(arrow.ListLikeType); ok {
+		return calcLeafCount(dt.Elem())
+	}
```

Comment:
could we further simply this by doing 

```go
if dt, ok := dt.(arrow.NestedType); ok {
    nleaves := 0
    for _, f := range dt.Fields() {
        nleaves += calcLeafCount(f.Type)
    }
    return nleaves
}
```
Which will cover struct types in addition to the list/map types?

Reply:
Done in 9ce73c4a0e8ccbc0c1f0fdc632666df4dc947750.
I removed the panic on the union types then (as they are nested, BTW).